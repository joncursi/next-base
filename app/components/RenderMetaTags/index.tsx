/**
 * @prettier
 */

import * as React from 'react';
import Head from 'next/head';

import NAMES from '~/constants/names';

interface Props {
  description?: string;
  imageUrl?: string;
  keywords?: string[];
  noIndex?: boolean;
  title?: string;
}

const RenderMetaTags: React.FC<Props> = ({
  description,
  imageUrl = '/static/img/meta-default.png',
  keywords,
  noIndex,
  title = NAMES.APP,
}: Props): React.ReactElement<Props> => (
  <Head>
    {/* Descriptions */}
    {description && (
      <>
        {/* Search Engines */}
        <meta content={description} name="description" />

        {/* Facebook Open Graph */}
        <meta content={description} property="og:description" />

        {/* Twitter Cards */}
        <meta content={description} name="twitter:text:description" />
      </>
    )}

    {/* Images */}
    {imageUrl && (
      <>
        {/* Facebook Open Graph */}
        <meta content={imageUrl} property="og:image" />

        <meta content={imageUrl} name="twitter:image" />
      </>
    )}

    {/* Keywords */}
    {keywords && (
      <>
        {/* Search Engines */}
        <meta content={[NAMES.APP, keywords].toString()} name="keywords" />
      </>
    )}

    {/* Titles */}
    {title && (
      <>
        {/* Browser Tab, Search Engines */}
        <title itemProp="name" lang="en">
          {title}
        </title>

        {/* Facebook Open Graph */}
        <meta content={title} property="og:title" />

        {/* Twitter Cards */}
        <meta content={title} name="twitter:title" />
      </>
    )}

    {/* Utility Pages */}
    {noIndex && <meta content="noindex,follow" name="robots" />}
  </Head>
);

export default RenderMetaTags;
