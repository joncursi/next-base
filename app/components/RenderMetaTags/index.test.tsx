/**
 * @prettier
 */

import * as React from 'react';
import renderer from 'react-test-renderer';

import RenderMetaTags from '.';

describe('RenderMetaTags', () => {
  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <RenderMetaTags
          description="Some description."
          imageUrl="https://joncursi.com"
          keywords={['keyword1', 'keyword2', 'keyword3']}
          title="Some Title"
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
