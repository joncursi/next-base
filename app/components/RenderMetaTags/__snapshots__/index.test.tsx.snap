// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderMetaTags matches the snapshot 1`] = `null`;
